FinnUI/Sources/Components/SearchFilterTags/SearchFilterTagsView.swift

```diff
@@ -0,0 +1,255 @@
+        1
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        searchFilterTags.count
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeue(SearchFilterTagCell.self, for: indexPath)
+        cell.configure(with: searchFilterTags[indexPath.item], icon: viewModel.removeTagIcon)
+        cell.delegate = self
+        return cell
+    }
+}
+
+// MARK: - UICollectionViewDelegate
+
+extension SearchFilterTagsView: UICollectionViewDelegate {
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let offset = scrollView.contentOffset.x
+        let newWidth = filterButtonWidthConstraint.constant - offset
+
+        if newWidth > maxFilterButtonWidth {
+            filterButtonWidthConstraint.constant = maxFilterButtonWidth
+        } else if newWidth < minFilterButtonWidth {
+            filterButtonWidthConstraint.constant = minFilterButtonWidth
+        } else {
+            filterButtonWidthConstraint.constant = newWidth
+            scrollView.contentOffset.x = 0
+        }
+
+        let sizeRange = maxFilterButtonWidth - minFilterButtonWidth
+        let visibleRatio = (filterButtonWidthConstraint.constant - minFilterButtonWidth) / sizeRange
+        let newAlpha: CGFloat = visibleRatio > 0.25 ? visibleRatio : 0
+        filterButtonView.updateLabel(withAlpha: newAlpha)
+    }
+}
+
+// MARK: - SearchFilterButtonViewDelegate
+
+extension SearchFilterTagsView: SearchFilterButtonViewDelegate {
+    func searchFilterButtonViewDidSelectFilter(_ searchFilterButtonView: SearchFilterButtonView) {
+        delegate?.searchFilterTagsViewDidSelectFilter(self)
+    }
+}
+
+// MARK: - SearchFilterTagCellDelegate
+
+extension SearchFilterTagsView: SearchFilterTagCellDelegate {
+    func searchFilterTagCellDidSelectRemove(_ cell: SearchFilterTagCell) {
+        guard let indexPath = collectionView.indexPath(for: cell) else {
+            return
+        }
+
+        UIImpactFeedbackGenerator(style: .medium).impactOccurred()
+
+        searchFilterTags.remove(at: indexPath.item)
+        collectionView.deleteItems(at: [indexPath])
+
+        delegate?.searchFilterTagsView(self, didRemoveTagAt: indexPath.item)
+    }
+
+    func searchFilterTagCellWasSelected(_ cell: SearchFilterTagCell) {
+        guard let indexPath = collectionView.indexPath(for: cell) else {
+            return
+        }
+        delegate?.searchFilterTagsView(self, didTapSearchFilterTagAt: indexPath.item)
+    }
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+
+extension SearchFilterTagsView: UICollectionViewDelegateFlowLayout {
+    public func collectionView(_ collectionView: UICollectionView,
+                               layout collectionViewLayout: UICollectionViewLayout,
+                               sizeForItemAt indexPath: IndexPath) -> CGSize {
```

Comment:
Should probably be formatted like this to be consistent our style:

```suggestion
    public func collectionView(
        _ collectionView: UICollectionView,
        layout collectionViewLayout: UICollectionViewLayout,
        sizeForItemAt indexPath: IndexPath
    ) -> CGSize {
```